File: 2024/1.two-sum.ts
/*
 * @lc app=leetcode id=1 lang=typescript
 *
 * [1] Two Sum
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
  const hashmap = new Map<number, number>();
  nums.forEach((num, i) => hashmap.set(target - num, i));
  for (let i = 0; i < nums.length; i++) {
    if (hashmap.has(nums[i]) && hashmap.get(nums[i]) !== i) {
      const index = hashmap.get(nums[i]);
      if (!index) {
        throw new Error('There must be a result')
      }
      return [i, index];
    }
  }
  throw new Error('There must be a result');
}
// @lc code=end

File: 2024/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i, num in enumerate(nums):
            if hashmap.get(num) is not None:
                return [hashmap[num], i]
            hashmap[target - num] = i

# @lc code=end

